client/keys/list.go

```diff
@@ -26,9 +26,13 @@ along with their associated name and address.`,
 }
 
 func runListCmd(cmd *cobra.Command, _ []string) error {
+	keyringDir, _ := cmd.Flags().GetString(flags.FlagKeyringDir)
+	if keyringDir == "" {
+		keyringDir, _ = cmd.Flags().GetString(flags.FlagHome)
+	}
```

Comment:
We don't need to do these calls in each command. Can we not use the `Context`?

Reply:
> We don't need to do these calls in each command. Can we not use the `Context`?

The `keys *` commands don't appear to have a Context.  What do you have in mind?

Reply:
Or at least they don't use the `clientCtx` at all and I was just following how `--keyring-backend` was implemented.